usaspending_api/common/helpers/s3_helpers.py

```diff
@@ -87,15 +89,47 @@ def multipart_upload(bucketname, regionname, source_path, keyname):
     transfer.upload_file(source_path, bucketname, Path(keyname).name, extra_args={"ACL": "bucket-owner-full-control"})
 
 
-def download_s3_object(bucket_name: str, key: str, file_path: str, region_name: str = settings.USASPENDING_AWS_REGION):
```

Comment:
Why add the knowledge of retry logic here. I think it'd be better to add retry logic to the code that calls this method. For example, my covid-19 download work doesn't need this retry logic

Reply:
Essentially, you've taken this very simple function and made it more complex solely for the sake of table from Delta Lake to Postgres copy.

Reply:
I thought this function could benefit from retry logic. I think it's a pretty common case that if we're trying to download a file from S3, we'd want the ability to retry, even in your Covid-19 download code.

I knew that code used this function, and considered setting the `retry_count` to `0`, but ultimately decided to let it use the default of `3` because we might as well let it retry.

Reply:
Ok, I'm for it.